scripts/globals/weaponskills.lua

```diff
@@ -959,3 +959,62 @@ function takeWeaponskillDamage(defender, attacker, params, finaldmg, slot, tpHit
 
     return finaldmg;
 end
+
+-- Params should have the following members:
+-- params.power.lv1: Base value for AM power @ level 1
+-- params.power.lv2: Base value for AM power @ level 2
+-- params.power.lv3: Base value for AM power @ level 3
+
+-- params.power.lv1_inc: How much to increment at each power level
+-- params.power.lv2_inc: How much to increment at each power level
+
+-- params.subpower.lv1: Subpower for level 1
+-- params.subpower.lv2: Subpower for level 2
+-- params.subpower.lv3: Subpower for level 3
+
+-- params.duration.lv1: Duration for AM level 1
+-- params.duration.lv2: Duration for AM level 2
+-- params.duration.lv3: Duration for AM level 3
+function applyAftermathEffect(player, tp, params)
+    if (params == nil) then
+        params = initAftermathParams()
+    end
+
+    local apply_power = 0
+    if (tp == 3000) then
+        player:addStatusEffect(EFFECT_AFTERMATH_LV3, params.power.lv3, 0, 
+        params.duration.lv3, 0, params.subpower.lv3)
+    elseif (tp >= 2000) then
+        apply_power = params.power.lv2 + (params.power.lv2_inc * ((tp - 2000) / 100))
+        player:addStatusEffect(EFFECT_AFTERMATH_LV2, apply_power, 0, 
```

Comment:
no, all that does is basically revert starting from 24 (the only time you could ever hit 20 is if you were at exactly 2000 tp).  Chances are it's really 20-59, but "20-60" is just for simplicity.  
I think the step increments should be smooth rather than jump by inc, though. (2975 tp = 59 attack).
applied_power = power + ((tp - 2000) / (100 / inc))